KubeArmor/BPF/system_monitor.c

```diff
@@ -429,50 +468,62 @@ static __always_inline u32 remove_pid_ns()
     return 0;
 }
 
-static __always_inline u32 skip_syscall()
+static __always_inline u32 skip_syscall(const u32 scope)
 {
+    struct outer_key okey;
+    u32 decision = ignore_syscall;
     struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+    u32 pid_ns = get_task_pid_ns_id(task);
+    get_outer_key(&okey, task, &pid_ns);
 
-#if defined(MONITOR_HOST)
+    u32 *ns_visibility = bpf_map_lookup_elem(&kubearmor_visibility, &okey);
 
-    u32 pid_ns = get_task_pid_ns_id(task);
-    if (pid_ns != PROC_PID_INIT_INO) {
-        return 1;
+    if (!ns_visibility) {
+        return trace_syscall;
+    }
+
+    u32 *on_off_switch = bpf_map_lookup_elem(ns_visibility, &scope);
+
+    if (!on_off_switch) {
+        return trace_syscall;
     }
 
+    if (*on_off_switch){
+        return ignore_syscall;
+    }
+#if defined(MONITOR_HOST)
+
```

Comment:
Do we need this anymore? I think we use 0,0 to signify that right? or will it be handled in a separate PR?